neuro_san/internals/graph/persistence/registry_manifest_restorer.py

```diff
@@ -129,11 +128,36 @@ def restore(self, file_reference: str = None):
                 file_of_class = FileOfClass(manifest_file)
                 manifest_dir: str = file_of_class.get_basis()
                 registry_restorer = AgentToolRegistryRestorer(manifest_dir)
-                tool_registry: AgentToolRegistry = registry_restorer.restore(file_reference=use_key)
+                try:
+                    tool_registry: AgentToolRegistry = registry_restorer.restore(file_reference=use_key)
+                except FileNotFoundError as exc:
+                    self.logger.error("Failed to restore registry item %s - %s", use_key, str(exc))
+                    tool_registry = None
                 if tool_registry is not None:
```

Comment:
In case agent file is referenced in manifest file, but doesn't actually exists (because we just deleted it or something),
handle FileNotFoundError right here, otherwise it will crash updater thread and we lose periodic updates.